File: src/Orchestrator/src/Application/General/Queries/GetContractsStates/VM/GetContractsStatesRvm.cs
using Application.General.Queries.GetContractsStates.QM;

namespace Application.General.Queries.GetContractsStates.VM;

public class GetContractsStatesRvm
{
    public required TruQuestContractInfoQm TruQuestInfo { get; init; }
    public required IEnumerable<string> WhitelistedWalletAddresses { get; init; }
    public required IEnumerable<UserVm> Users { get; init; }
    public required IEnumerable<SubjectVm> Subjects { get; init; }
}

File: src/frontend/lib/admin/models/vm/get_contracts_states_rvm.dart
import 'subject_vm.dart';
import 'truquest_contract_info_vm.dart';
import 'user_vm.dart';

class GetContractsStatesRvm {
  final TruQuestContractInfoVm truQuestInfo;
  final List<String> whitelistedWalletAddresses;
  final List<UserVm> users;
  final List<SubjectVm> subjects;

  GetContractsStatesRvm.fromMap(Map<String, dynamic> map)
      : truQuestInfo = TruQuestContractInfoVm.fromMap(map['truQuestInfo']),
        whitelistedWalletAddresses = List<String>.unmodifiable(map['whitelistedWalletAddresses'] as List<dynamic>),
        users = List.unmodifiable(
          (map['users'] as List<dynamic>).map((submap) => UserVm.fromMap(submap)),
        ),
        subjects = List.unmodifiable(
          (map['subjects'] as List<dynamic>).map((submap) => SubjectVm.fromMap(submap)),
        );
}
